README.md

```diff
@@ -74,10 +74,28 @@ proc waitFor*[T](fut: Future[T]): T =
   return fut.read()
 ```
 
+### Async procedures and methods
+
+The `{.async.}` pragma will transform a procedure (or a method) returning a
+specialised `Future` type into a closure iterator. If there is no return type
+specified, a `Future[void]` is returned.
+
+```nim
+proc p() {.async.} =
+  await sleepAsync(100.milliseconds)
+
+echo p().type # prints "Future[system.void]"
+```
+
+Whenever `await` is encountered inside an async procedure, control is passed
+back to the dispatcher for as many steps as it's necessary for the awaited
+future to complete (or be cancelled). `await` calls the equivalent of
```

Comment:
complete, failed or be cancelled.

Reply:
Chronos defines "completion" as success, failure or cancellation:
https://github.com/status-im/nim-chronos/blob/f856c885facfb45074b95c5df42f6468f2cc86da/chronos/asyncfutures2.nim#L147-L154

Reply:
Terminology clarified.